src/lib/es5.d.ts

```diff
@@ -430,7 +430,7 @@ interface String {
     /**
      * Replaces text in a string, using a regular expression or search string.
      * @param searchValue A string to search for.
-     * @param replaceValue A string containing the text to replace for every successful match of searchValue in this string.
+     * @param replaceValue A string containing the text to replace. When the searchvalue is a string, only the first match is replaced. If the searchValue is a Regexp, all matches are replaced if the g flag is set. Otherwise only the first one is.

```

Comment:
There's an "interesting" aspect where `"aaa-a".replace(/a/gy, "x")` is `xxx-a`. I don't know how detailed we want it to be, but "all matches are replaced if the g flag is set" is technically not correct.

Reply:
MDN's documentation for the sticky flag says that it ignores the global flag. If that's true, then I think it's fine to document it as-is, since you'll presumably find out about `y` long after `g` and hopefully read about the override.

However, is that consistent with your example? If there's some kind of interaction, it *might* be worth documenting.

Reply:
There's an open issue for it: https://github.com/mdn/content/issues/1454 I just haven't gotten around to fix it.